notebooks/samples/LightGBM Overview.ipynb

```diff
@@ -0,0 +1,548 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(model.getFeatureImportances())"
+   ]
+  },
+  {
+   "source": [
+    "#### Model Prediction"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scoredData = model.transform(test)\n",
+    "display(scoredData)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mmlspark.train import ComputeModelStatistics\n",
+    "metrics = ComputeModelStatistics(evaluationMetric='regression',\n",
+    "                                 labelCol='label',\n",
+    "                                 scoresCol='prediction') \\\n",
+    "            .transform(scoredData)\n",
+    "display(metrics)"
+   ]
+  },
+  {
+   "source": [
+    "## LightGBM Ranker"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "#### Read dataset"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pyspark.sql.functions import monotonically_increasing_id\n",
+    "from pyspark.sql.functions import explode"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df1 = spark.read.format('libsvm') \\\n",
+    "    .load(\"wasbs://publicwasb@mmlspark.blob.core.windows.net/lightGBMRanker_rank_train.libsvm\") \\\n",
+    "    .withColumn('iid', monotonically_increasing_id())\n",
+    "display(df1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from pyspark.sql.types import ArrayType, LongType\n",
+    "def create_rows(value, index):\n",
+    "    arr = np.zeros([value])\n",
+    "    arr.fill(index)\n",
+    "    return arr.astype('int').tolist()\n",
+    "create_rows_udf = udf(create_rows, ArrayType(LongType()))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query_col = 'query'\n",
+    "label_col = 'labels'\n",
+    "# read in CSV file\n",
+    "df2 = spark.read.format('csv').option('inferSchema', True) \\\n",
+    "    .load(\"wasbs://publicwasb@mmlspark.blob.core.windows.net/lightGBMRanker_rank_train_query.csv\") \\\n",
+    "    .withColumn('index', monotonically_increasing_id()) \\\n",
+    "    .withColumn(query_col, explode(create_rows_udf('_c0', 'index'))) \\\n",
+    "    .withColumn('iid', monotonically_increasing_id()) \\\n",
+    "    .drop('_c0', 'index') \\\n",
+    "    .join(df1, 'iid').drop('iid') \\\n",
+    "    .withColumnRenamed('label', label_col)"
```

Comment:
Can we save the result of this to parquet in the public blob then read it in so the demo is smaller?